streambed/src/lib.rs

```diff
@@ -23,16 +24,14 @@ pub mod crypto;
 pub mod delayer;
 pub mod secret_store;
 
-/// Read the secret store's secret id associated with a service's role.
-/// We read the secret from stdin to avoid requiring secrets being
-/// persisted elsewhere.
-pub async fn secret_id_from_stdin() -> Result<String, io::Error> {
-    let mut ss_secret_id = String::new();
+/// Read a line from stdin.
+pub async fn read_line_from_stdin() -> Result<String, io::Error> {
+    let mut line = String::new();
     let mut reader = BufReader::new(io::stdin());
-    reader.read_line(&mut ss_secret_id).await?;
-    let len = ss_secret_id.trim_end_matches(&['\r', '\n'][..]).len();
-    ss_secret_id.truncate(len);
-    Ok(ss_secret_id)
+    reader.read_line(&mut line).await?;
+    let len = line.trim_end_matches(&['\r', '\n'][..]).len();
+    line.truncate(len);
+    Ok(line)
 }
```

Comment:
This function waits indefinitely and never times out, is that what we want?

Reply:
Yes. We cannot proceed without the keys.